File: gateway_system_v5_python/src/services/route_service.py
from src.models.route import Route


class RouteService:
    def __init__(self):
        self.routes: dict[str, Route] = {}

    def get_all(self) -> list[Route]:
        return list(self.routes.values())

    def get(self, id: str) -> Route | None:
        return self.routes.get(id)

    def add(self, route: Route) -> Route:
        self.routes[route.id] = route
        return route

    def remove(self, id: str) -> None:
        self.routes.pop(id, None)

    def find_by_path(self, path: str) -> Route | None:
        for route in self.routes.values():
            if route.enabled and path.startswith(route.path):
                return route
        return None

File: gateway_system_v3_java/src/main/java/com/gateway/service/RouteService.java
package com.gateway.service;

import com.gateway.model.Route;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;

@Service
public class RouteService {

    private final Map<String, Route> routes = new ConcurrentHashMap<>();

    public List<Route> getAllRoutes() {
        return new ArrayList<>(routes.values());
    }

    public Optional<Route> getRoute(String id) {
        return Optional.ofNullable(routes.get(id));
    }

    public Route addRoute(Route route) {
        routes.put(route.getId(), route);
        return route;
    }

    public void removeRoute(String id) {
        routes.remove(id);
    }

    public Optional<Route> findRouteByPath(String path) {
        return routes.values().stream()
                .filter(Route::isEnabled)
                .filter(r -> path.startsWith(r.getPath()))
                .findFirst();
    }

    public void reloadRoutes() {
        // Placeholder for configuration reload
    }
}
